src/js/App.js

```diff
@@ -1,40 +1,47 @@
 /* eslint-disable class-methods-use-this */
-import ProductManageMenu from './components/ProductManageMenu/index.js';
-import VendingMachineManageMenu from './components/VendingMachineManageMenu/index.js';
 import { SELECTOR } from './constants/selector.js';
 import { $ } from './utils/dom.js';
 
+import './components/ProductManageMenu/index.js';
+import './components/VendingMachineManageMenu/index.js';
+import './components/ProductPurchaseMenu/index.js';
+import { routes } from './route/routes.js';
+import { navigate } from './route/navigate.js';
+
 const CATEGORY = {
   PRODUCT_MANAGE_MENU: 'product-manage-menu',
   VENDING_MACHINE_MANAGE_MENU: 'vending-machine-manage-menu',
   PRODUCT_PURCHASE_MENU: 'product-purchase-menu',
 };
 
-class App {
+class App extends HTMLElement {
   #state = {
     category: CATEGORY.PRODUCT_MANAGE_MENU,
   };
 
-  #productManageMenu = new ProductManageMenu();
-
-  #vendingMachineManageMenu = new VendingMachineManageMenu();
-
-  init() {
+  connectedCallback() {
     this.#render();
     this.#bindEvents();
+    navigate('/product-manage-menu');
+  }
+
+  #findMatchedRoute() {
+    return routes.find((route) => route.path.test(window.location.pathname));
+  }
+
+  #route() {
+    this.innerHTML = this.#findMatchedRoute().element;
   }
```

Comment:
```suggestion
  #route() {
    const currentRoute = routes.find((route) => route.path.test(window.location.pathname));
    this.innerHTML = currentRoute.element;
  }
```
(마이너) 혹시 `findMatchedRoute`를 메서드로 분리해주신 이유가 있으신걸까요?
재활용될 여지가 없을 것 같아서 메서드로 분리하지 않는편이 더 좋아보여요.

Reply:
생각하기에는 '깔끔하게 분리하자' 라는 의도로 했던 것 같은데 큰 의미가 없이 메서드로 분리가 된 것 같습니다.
제안주신 코드처럼 `route` 메서드 내부에서 작성하는 게 괜찮을 것 같습니다!